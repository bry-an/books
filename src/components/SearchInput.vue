<template>
  <div>
    <h4>Search by author, title, or subject</h4>
    <v-form>
      <v-text-field label="Search" placeholder="e.g. Hemmingway" v-model="userSearch"></v-text-field>
    </v-form>
    <v-btn @click="submitSearch" id="search-submit-btn" color="primary">Search</v-btn>
    <!-- <button @click="submitSearch">Submit</button> -->
  </div>
</template>

<script>
import axios from "axios";
export default {
  name: "SearchInput",
  data() {
    return {
      userSearch: "",
      searchResults: [],
      numOfResults: null
    };
  },
  methods: {
    async submitSearch() {
      console.log("submit search!");
      const queryUrl = `http://openlibrary.org/search.json?q=${
        this.userSearch
      }`;
      const result = {
        results: [],
        numOfResults: 0
      };

      await axios.get(queryUrl).then(response => {
        result.results = [...response.data.docs];
        result.numOfResults = response.data.numFound;
      });
      this.$emit("set-results", result);
    }
  }
};
</script>

<style lang="sass" scoped>
input[type="text"]
    display: block
    width: 200px
    margin: 8px 0
    border: 0.5px solid
</style>