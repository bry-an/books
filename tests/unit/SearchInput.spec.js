import { mount } from "@vue/test-utils"
import Vuetify from "vuetify"
import SearchInput from "../../src/components/SearchInput.vue"
import Vue from "vue"
Vue.use(Vuetify)
import flushPromises from "flush-promises"

describe("SearchInput", () => {
    let build
    build = () => {
        const wrapper = mount(SearchInput)
        return {
            wrapper,
            input: () => wrapper.find("input[type='text']"),
            button: () => wrapper.find("button")
        }
    }
    it("renders the component", () => {
        const { wrapper } = build()
        expect(wrapper.html()).toMatchSnapshot()
    })


    // it("fetches book data from api", async () => { const { wrapper, button, input } = build()

    //     console.log(wrapper.mocks)

    //     input().setValue("hemmingway")
    //     button().trigger('click')
    //     await flushPromises()
    //     expect(wrapper.emitted("set-results")).toBeTruthy()
    // })
})


/*
        const mockResponse = {}
        const mockJsonPromise = Promise.resolve(mockResponse)
        const mockFetchPromise = Promise.resolve({
            json: () => mockJsonPromise
        })
        */