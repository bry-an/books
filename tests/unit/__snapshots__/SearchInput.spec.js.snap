// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput renders the component 1`] = `
<div>
  <h4>Search by author, title, or subject</h4>
  <form novalidate="novalidate" class="v-form">
    <div class="v-input v-text-field v-text-field--placeholder theme--light">
      <div class="v-input__control">
        <div class="v-input__slot">
          <div class="v-text-field__slot"><label aria-hidden="true" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Search</label><input aria-label="Search" type="text" placeholder="e.g. Hemmingway"></div>
        </div>
        <div class="v-text-field__details">
          <div class="v-messages theme--light">
            <div class="v-messages__wrapper"></div>
          </div>
        </div>
      </div>
    </div>
  </form> <button type="button" class="v-btn theme--light primary" id="search-submit-btn">
    <div class="v-btn__content">Search</div>
  </button>
</div>
`;
